jdaviz/app.py

```diff
@@ -1013,21 +1013,18 @@ def add_data_to_viewer(self, viewer_reference, data_path,
         viewer_reference : str
             The reference to the viewer defined with the ``reference`` key
             in the yaml configuration file.
-        data_path : str
-            Either the data filename or the Glue data label found in the ``DataCollection``.
+        data_label : str
+            The Glue data label found in the ``DataCollection``.
         clear_other_data : bool
             Removes all other currently plotted data and only shows the newly
             defined data set.
-        ext: str
```

Comment:
This was added by @havok2063 in #158 . We don't want to be removing some feature that MAST needs, so let's see what he has to say.

Reply:
I don't think this works in main (through `add_data_to_viewer`), although the original intent in 2020 likely was different 🤷

Reply:
I dunno what weird stuff others are doing, so would be a relieve if Brian C can bless this patch.

Reply:
Weird stuff coming atcha!!   Originally this function only accepted a data label as input, and back in those days, data labels were random UUIDs, and hard to identify / extract from the application for loading.  Or data labels were of the format "filename[EXT], i.e. `jw...[SCI]`, without the extension, forcing string manipulation.  I didn't like any of that and wanted the ability to just specify a string filename, and the name of the file extension I wanted to load.  The code would create the data label from the relevant input.    See orig threads https://github.com/spacetelescope/jdaviz/pull/158 and https://github.com/spacetelescope/jdaviz/issues/112.  The updated functions allowed the user to input either a data_label, preserving the original functionality, or a filename + extension name/number.  

I would like to retain the functionality of a user supplying a filepath and an extension name or number as input into `add/remove_data_from_viewer`, as I still think that's easier for users to do rather than creating data labels themselves.  How that would work with the new default data labeling mechanism I leave up to y'all to figure out.  
I think there's a use case for programmatically creating viewers, and wanting to load data quickly from a single file into different loaders, without all the string manipulation.  

All that being said, I'll note that we do not use these functions in production in MAST.   At some point we switched to using the helper's `load_data` methods.   I think because I could give it only a filename and it was happy.

Reply:
Thanks for the context @havok2063!  Just from reading the code before this PR, I don't think passing a filename to `add_data_to_viewer` worked in main, at least not in the recent past.  So I think my vote would be to move forward with this as-is and open a follow-up ticket/issue to consider either proper support for passing data/filename directly to `add_data_to_viewer` or perhaps more cleanly support passing a list of viewer names/ids to `app.load_data` so they can be immediately loaded in one line.

Reply:
Also, you should definitely check out the improvements introduced by #1672; no more uuids!